aws/cloudformation/iam.yml.erb

```diff
@@ -147,6 +147,19 @@ Resources:
               - dynamodb:GetItem
             Resource: !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/dcdo_production"
 
+  # Policy to deny the ability to delete S3 buckets. Deletion must be performed by an Admin
+  DenyS3DeleteBucketPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      ManagedPolicyName: DenyS3Delete
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Deny
+            Action:
+              - s3:DeleteBucket
+            Resource: "arn:aws:s3:::cdo-*"
```

Comment:
Are there buckets outside this prefix we should include?

Should we just use '*' and deny for all?

Reply:
Is there a way we can refine this to just production buckets (maybe using the `environment` tag)? I’m thinking about Javabuilder, where each Stack has its own buckets to store user generated content and an engineer and their CodePipeline needs to be able to delete the buckets when they de-provision their development Stack. While we currently use buckets that are shared across all environments to store student project code & assets, we hopefully will implement separate buckets per Stack in the future.

Reply:
Related PR on that topic - tagging might be a good solution here.
https://github.com/code-dot-org/javabuilder/pull/394